src/category_theory/monad/adjunction.lean

```diff
@@ -84,8 +85,46 @@ def monad.comparison_forget (h : L ⊣ R) :
 
 lemma monad.left_comparison (h : L ⊣ R) : L ⋙ monad.comparison h = h.to_monad.free := rfl
 
-instance [faithful R] (h : L ⊣ R) :
-  faithful (monad.comparison h) :=
+/-- A technical lemma which helps show uniqueness of the comparison functor. -/
+lemma monad.comparison_unique_aux {L : C ⥤ D} {R : D ⥤ C} {h : L ⊣ R}
+  {K : D ⥤ h.to_monad.algebra}
+  {i : K ⋙ h.to_monad.forget ≅ R}
+  (hK : ∀ Y,
+    R.map (L.map (i.hom.app (L.obj (R.obj Y)))) ≫ R.map (h.counit.app (L.obj (R.obj Y))) =
+      (K.obj (L.obj (R.obj Y))).a ≫ i.hom.app (L.obj (R.obj Y))) :
+  ∀ Y, R.map (L.map (i.hom.app Y)) ≫ R.map (h.counit.app Y) = (K.obj Y).a ≫ i.hom.app Y :=
+begin
+  intro Y,
+  haveI : split_epi (R.map (h.counit.app Y)) := ⟨h.unit.app _, h.right_triangle_components⟩,
+  rw [←is_iso.eq_inv_comp, ←cancel_epi (R.map (L.map (R.map (h.counit.app Y))))],
+  dsimp only [functor.id_obj],
+  have : R.map (L.map (K.map (h.counit.app Y)).f) ≫ (K.obj Y).a =
+          (K.obj (L.obj (R.obj Y))).a ≫ (K.map (h.counit.app Y)).f := (K.map (h.counit.app Y)).h,
+  rw [←R.map_comp, h.counit_naturality, ←functor.map_inv, ←functor.map_inv, ←R.map_comp_assoc,
+    ←L.map_comp, ←nat_iso.is_iso_inv_app, is_iso.iso.inv_hom, i.inv.naturality, L.map_comp,
+    R.map_comp, R.map_comp, assoc, ←is_iso.inv_comp_eq, ←functor.map_inv, ←functor.map_inv,
+    ←nat_iso.is_iso_inv_app, is_iso.iso.inv_inv],
+  dsimp only [functor.comp_obj, functor.comp_map],
+  rw [reassoc_of hK, monad.forget_map, reassoc_of this, ←i.hom.naturality],
+  refl,
+end
+
+/-- Given an adjunction `h : L ⊣ R` and an functor `K` from `D` to the category of algebras on the
+monad induced by `h`, if `K ⋙ monad.forget _` is isomorphic to `R` and the isomorphism commutes
+with the monad on objects of the from `L (R Y)`, produce an isomorphism from `K` to the comparison
+functor. -/
+@[simps]
+def monad.comparison_unique {L : C ⥤ D} {R : D ⥤ C} {h : L ⊣ R} {K : D ⥤ h.to_monad.algebra}
+  (i : K ⋙ h.to_monad.forget ≅ R)
+  (hK' : ∀ (Y : D),
+    R.map (L.map (i.hom.app (L.obj (R.obj Y)))) ≫ R.map (h.counit.app (L.obj (R.obj Y))) =
+      (K.obj (L.obj (R.obj Y))).a ≫ i.hom.app (L.obj (R.obj Y))):
```

Comment:
This condition is somewhat difficult to parse, since you've written it in what looks like simp normal form. Is the intention that `tidy` is more likely to use this lemma in this form? Would that actually work?

I don't know whether it would be better to write something that looks closer to the assumption `w` in `monad.algebra.iso_mk`, for example. What do you think?

Reply:
Yeah it's a tricky one, this condition is actually the same as the assumption `w` but restricted to elements in the image of R >>> L, the point being that because of `monad.comparison_unique_aux` we can self-strengthen the result to everything in `D`. It could be nice to write this in a more convenient way, do you have any ideas?